fetch.go

```diff
@@ -38,7 +37,7 @@ func newFetchProtocol(ctx context.Context, host host.Host, getData getValue) *fe
 }
 
 func (p *fetchProtocol) receive(s network.Stream, getData getValue) {
-	defer helpers.FullClose(s)
+	defer s.Close()
```

Comment:
Do we need to add a reset to this function if `writeMsg` returns an error like you added in the Fetch function? It looks like it doesn't really matter since `writeMsg` does a Reset anyway if it errors, but we should be consistent.

Perhaps add a line documenting whether `read/writeMsg` Reset if there's an error or whether the caller should.

Sorry it's a bit messy, guess I didn't know as much about how to use the stream interfaces when I wrote this code 😬.

Reply:
Eh, I've just removed the reset from writeMsg. It's generally assumed that the caller will handle closing the stream.